### DISCUSSION TITLE
Does it support Ergo erpc:call/erpc:cast to Erlang node?

### DISCUSSION THREAD

**Discussion by leonlee2013** [ergo-services/ergo#126] [Q&A] [answered]
Ergo rpc:call/erpc:cast to Erlang node is supported, the code is as follows:

https://github.com/ergo-services/ergo/blob/master/gen/server.go
```go
// CallRPCWithTimeout evaluate rpc call with given node/MFA and timeout
func (sp *ServerProcess) CallRPCWithTimeout(timeout int, node, module, function string, args ...etf.Term) (etf.Term, error) {
	lib.Log("[%s] RPC calling: %s:%s:%s", sp.NodeName(), node, module, function)

	message := etf.Tuple{
		etf.Atom("call"),
		etf.Atom(module),
		etf.Atom(function),
		etf.List(args),
		sp.Self(),
	}
	to := ProcessID{"rex", node}
	return sp.CallWithTimeout(to, message, timeout)
}

// CastRPC evaluate rpc cast with given node/MFA
func (sp *ServerProcess) CastRPC(node, module, function string, args ...etf.Term) error {
	lib.Log("[%s] RPC casting: %s:%s:%s", sp.NodeName(), node, module, function)
	message := etf.Tuple{
		etf.Atom("cast"),
		etf.Atom(module),
		etf.Atom(function),
		etf.List(args),
	}
	to := ProcessID{"rex", node}
	return sp.Cast(to, message)
}
```

But I didn't find any code or api related to erpc:call/erpc:cast to Erlang node.
https://www.erlang.org/blog/otp-23-highlights/#distributed-spawn-and-the-new-erpc-module

**Comment by halturin** [comment, marked as answer]
It doesn't support the new erpc way of communication yet. I would strongly recommend using the OTP-way, because RPC in erlang has performance issues and very limited features.

**Reply by leonlee2013** [reply]
Thanks for your reply!
Yes, many processes make rpc: calls to nodes at the same time and rex can become a bottleneck.
But erpc does not use rex at all and lets the newly generated process decode the parameters of the call, thus avoiding some unnecessary data replication that occurred in the "old" implementation.
So erpc solves this problem and has much higher performance than rpc. [Erlang/OTP24 has replaced rpc with erpc by default.](https://github.com/erlang/otp/blob/maint-24/lib/kernel/src/rpc.erl#:~:text=%3FTIMEOUT_TYPE.-,%2Difdef(SERVER_SIDE_ERPC_IS_MANDATORY).,%2Dendif.%20%25%25%20!%20defined%20SERVER_SIDE_ERPC_IS_MANDATORY,-%2Dspec%20block_call) Will Ergo have any plans for erpc support?

**Reply by halturin** [reply]
>Will Ergo have any plans for erpc support?

it's not a priority task now, but we are open to contributions.
